### ISSUE TITLE
Shares moving back to default "landing" dir 

### ISSUE THREAD

**Issue by mihaig11** [owncloud/core#12385]
### Steps to reproduce
I want three users to share files with each other, so:
1. create three users, user1, user2, user3
2. log in with user1 and create a folder "files from user2" (where files shared by user2 will be kept) and another folder "files from user3" (where files shared by user3 will be kept)
3. log in with user2 and share files/folders with user1. These will land in the root of user1 (as it should be, by default).
4. log in again with user1 and move the files/folders shared by user2 from root into "files from user2"
5. wait until files/folders in "files from user2" move back by themselves into the default "landing" folder for shared files.

### Expected behaviour
The files/folders shared by user2 and manually moved into "files from user2" (by user1) should always stay into "files from user2" folder

### Actual behaviour
The files/folders shared by user2 and manually moved into "files from user2" (by user1) move back into the home folder of user1 (the default "landing" folder for all files/folders) (unfortunately there is no clue in the logs about it and  I cannot reproduce it)

### Server configuration
**Operating system**:
Centos 7

**Web server:**
Apache 2.4

**Database:**
mariaDB

**PHP version:**
5.4.16

**ownCloud version:** (see ownCloud admin page)
7.0.3 (I had the same problem in 7.0.2)

**Updated from an older ownCloud or fresh install:**
from 7.0.2
**List of activated apps:**
activity
Deleted Files
Full Text Search
Mail Template Editor
Share Files
Text Editor
Versions

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'xx',
  'passwordsalt' => 'xx',
  'trusted_domains' =>
  array (
    0 => 'cloud.tpf.be',
  ),
  'datadirectory' => '/LocalDir/data',
  'dbtype' => 'mysql',
  'version' => '7.0.3.4',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xx',
  'dbpassword' => 'xx',
  'installed' => true,
  'forcessl' => true,
  'logtimezone' => 'Europe/Brussels',
  'theme' => 'tpf',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'smtp.tpf.be',
  'mail_smtpport' => '25',
  'loglevel' => '0',
  'mail_from_address' => 'cloud',
  'mail_domain' => 'tpf.be',
  'maintenance' => false,
);

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

### Client configuration
**Browser:** IE 10

**Operating system:** Windows 7 

#### Web server error log
 no errors in that log

#### ownCloud log (data/owncloud.log)
no errors in that log
#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by PVince81** [comment]
@mihaig11 are you still seeing this with 7.0.5 ?
AFAIK there were some sharing fixes that might help

**Comment by mihaig11** [comment]
@PVince81 I am not experiencing the issue anymore, so far, so good.
Thanks

**Comment by PVince81** [comment]
@mihaig11 thanks for the update. I'll close this ticket then.
If you see it happening again, feel free to reopen.
Thanks.

**Comment by mihaig11** [comment]
this is still a bug, I found a way to reproduce it with the owncloud virtual appliance downloaded from bitnami today  (OC 8.2.2).
Create 4 users: user1-4 and two groups group1-2. Put user1and user2 into group1 and user3and4 into group2.
login with user1. create a folder "foldergroup1" and share it read ony with group2 and modify for group1
login with user3. create a folder "foldergroup2" and share it read ony with group1 and modify for group2. create a folder "testfolder" and put "foldergroup1" and  "foldergroup2" in it.
login with all other users, create a folder "testfolder" and put "foldergroup1" and  "foldergroup2" in it.So all users should have a folder "testfolder" with "foldergroup1" and  "foldergroup2" in it.
log back with ocadmin and delete user4, for example.
log back with user1, for ex, and see that "foldergroup1" and/or  "foldergroup2" moved back into the root (the default "landing" folder).

**Comment by PVince81** [comment]
@rullzer

**Comment by rullzer** [comment]
assigning to myself so I don't forget to look into it.

@mihaig11 thanks for the detailed steps!

**Comment by rullzer** [comment]
Ok I can confirm this on master as well.

For some weird reason all the user specific group shares are deleted in the last step.
Debug time!

**Comment by rullzer** [comment]
Ok I think I got it... patch is almost trivial. PR incomming... and will be backported!

**Comment by rullzer** [comment]
Fix in #22143
